### PR TITLE
Changed flash layout.

### PR DESCRIPTION
**Pull request by vliedel**
- Changed flash layout:
  - P2p dfu page is now before microapp pages.
  - Decreased fds reserved.
- Changed default config:
  - Enable gpiote.
  - Enable microapps.
- Fixed cs_MemoryLayout.
- Updated memory doc.

### CODE REVIEW

File: config/nrf52840/CMakeBuild.config [hunk 1]
```diff
@@ -117,27 +117,34 @@ SOFTDEVICE_MINOR=2
...
+# On the nRF52832 the microapp size is 4 pages
+# On the nRF52840 we have more pages, so allow for 4 apps.
+# For now though, set number of pages to 4, as the code doesn't work yet for multiple microapps.
+# Page size is 0x1000.
+# 0xEF000 - (4+8+16)*0x1000 = 0xD3000
+FLASH_MICROAPP_BASE=0xD3000
+FLASH_MICROAPP_PAGES=4
+
+# We can use up to the P2P DFU page: FLASH_MICROAPP_BASE - 0x1000.
+# The maximum size of the app is then 0xD9000 - 0x1000 - 0x26000 = 0xB2000
```
Comment (ArrowAcrobatics) [inline config/nrf52840/CMakeBuild.config:135]: is this 0xB2000 still correct?

File: source/conf/cmake/CMakeBuild.config.default [hunk 2]
```diff
@@ -381,7 +381,7 @@ APPLICATION_START_ADDRESS=0x00026000
 # The start is the above start address.
 # The end is calculated from bootloader start: minus fds (8 pages), p2p dfu (1 page), and microapp (4 pages).
 # The size is the end minus the start. For the nRF52832 this is: 0x76000-(8+4+1)*0x1000-0x26000 = 0x43000
```
Comment (ArrowAcrobatics) [inline source/conf/cmake/CMakeBuild.config.default:383]: this comment needs an update I think

**Review by ArrowAcrobatics — COMMENTED** [review]
only added some minor comments, ok to merge after checking

File: config/nrf52840/CMakeBuild.config (hunk 1 above)
Comment (vliedel) [reply config/nrf52840/CMakeBuild.config:135]: Updated, there are too many comments with config values in them :(

File: source/conf/cmake/CMakeBuild.config.default (hunk 2 above)
Comment (vliedel) [reply source/conf/cmake/CMakeBuild.config.default:383]: updated

**Pull request merged**